metadata-db-app/int_test/cmr/metadata_db/int_test/concepts/force_delete_test.clj

```diff
@@ -66,3 +75,60 @@
        ;; tag association associated with other collection revision or whole collection is not deleted
        (util/is-tag-association-deleted? ta1 false)
        (util/is-tag-association-deleted? ta3 false)))))
+
+(deftest force-delete-service-with-associations
+  (let [coll (util/create-and-save-collection "REG_PROV" 1)
+        coll-concept-id (:concept-id coll)
+        svc-concept (util/create-and-save-service "REG_PROV" 1 2)
+        svc-concept-id (:concept-id svc-concept)
+        svc-assn (util/create-and-save-service-association
+                  coll svc-concept 1)
+        svc-assn-concept-id (:concept-id svc-assn)]
+    (testing "initial conditions"
+      ;; creation results as expected
+      (is (= 2 (:revision-id svc-concept)))
+      ;; service revisions in db
+      (is (= 2
+             (count (:concepts (util/find-concepts :service)))))
+      (is (= 200 (:status (util/get-concept-by-id-and-revision
+                           svc-concept-id 2))))
+      (is (= 200 (:status (util/get-concept-by-id-and-revision
+                           svc-concept-id 1))))
+      ;; make sure service association in place
+      (is (= coll-concept-id
+             (get-in svc-assn [:extra-fields :associated-concept-id])))
+      (is (= svc-concept-id
+             (get-in svc-assn [:extra-fields :service-concept-id])))
+      ;; make sure collection associations in place
+      ;; XXX this test originally just checked counts; when I saw the counts didn't
```

Comment:
@ygliuvt and I will be pairing on this today ... it will be an exploration for both of us, since it's not completely obvious which is the correct path to take.